src/main/java/io/wisetime/connector/testutils/FakeEntities.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2018 Practice Insight Pty Ltd. All Rights Reserved.
+ */
+
+package io.wisetime.connector.testutils;
+
+import com.github.javafaker.Faker;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import io.wisetime.generated.connect.Tag;
+import io.wisetime.generated.connect.TimeGroup;
+import io.wisetime.generated.connect.TimeRow;
+import io.wisetime.generated.connect.User;
+
+import static java.lang.String.format;
+
+/**
+ * @author shane.xie@practiceinsight.io
+ */
+public class FakeEntities {
+
+  private static final Faker FAKER = new Faker();
+  private static final String TAG_PATH = format("/{}/{}/", FAKER.lorem().word(), FAKER.lorem().word());
+
+  public TimeGroup randomTimeGroup() {
+    final List<TimeRow> timeRows = randomEntities(() -> randomTimeRow(), 1, 10);
+
+    return new TimeGroup()
+        .callerKey(FAKER.bothify("#?#?#?#?#?"))
+        .groupId(UUID.randomUUID().toString())
+        .description(FAKER.lorem().paragraph())
+        .totalDurationSecs(timeRows.stream().mapToInt(TimeRow::getDurationSecs).sum())
+        .groupName(FAKER.color().name())
+        .tags(randomEntities(() -> randomTag(TAG_PATH), 1, 3))
+        .user(randomUser())
+        .timeRows(timeRows)
+        .narrativeType(randomEnum(TimeGroup.NarrativeTypeEnum.class))
+        .durationSplitStrategy(randomEnum(TimeGroup.DurationSplitStrategyEnum.class));
+  }
+
+  public Tag randomTag(final String path) {
+    return new Tag()
```

Comment:
Unused variable `path`
```suggestion
    return new Tag()
        .path(path)
```